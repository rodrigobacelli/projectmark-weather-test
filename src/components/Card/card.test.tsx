import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { Card } from './';

describe('components/Card', () => {
  it('should render the Card children', () => {
    const result = render(<Card>Card Content</Card>);

    expect(result.getByText('Card Content')).toBeInTheDocument();
  });

  it('should accept classNames', () => {
    const result = render(<Card className="card-className">Card Content</Card>);

    expect(result.container.querySelector('.card-className')).toBeTruthy();
  });

  it('should render the Card Component', () => {
    const result = render(<Card className="card-className">Card Content</Card>);

    expect(result).toMatchSnapshot();
  });
});
