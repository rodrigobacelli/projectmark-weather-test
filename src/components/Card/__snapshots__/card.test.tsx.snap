// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/Card > should render the Card Component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="p-4 bg-gray-100 dark:bg-gray-800 rounded-lg ring shadow-xl ring-gray-900/5 card-className"
      >
        Card Content
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="p-4 bg-gray-100 dark:bg-gray-800 rounded-lg ring shadow-xl ring-gray-900/5 card-className"
    >
      Card Content
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
